gpiozero/pins/local.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import (
+    unicode_literals,
+    absolute_import,
+    print_function,
+    division,
+    )
+str = type('')
+
+import io
+import warnings
+
+try:
+    from spidev import SpiDev
+except ImportError:
+    SpiDev = None
+
+from . import SPI
+from .pi import PiFactory, PiPin
+from .spi import SPISoftwareBus
+from ..devices import Device, SharedMixin
+from ..output_devices import OutputDevice
+from ..exc import DeviceClosed, PinUnknownPi, SPIInvalidClockMode
+
+
+class LocalPiFactory(PiFactory):
+    """
+    Abstract base class representing pins attached locally to a Pi. This forms
+    the base class for local-only pin interfaces (:class:`RPiGPIOPin`,
+    :class:`RPIOPin`, and :class:`NativePin`).
+    """
+    pins = {}
+
+    def __init__(self):
+        super(LocalPiFactory, self).__init__()
+        self.spi_hardware_class = LocalPiHardwareSPI
+        self.spi_software_class = LocalPiSoftwareSPI
+        self.shared_spi_hardware_class = LocalPiHardwareSPIShared
+        self.shared_spi_software_class = LocalPiSoftwareSPIShared
```

Comment:
That could indeed be neater - perhaps keyed by a (hardware, shared) tuple of bools. I'll try it and see how it looks.